ksummarized/ksummarized.com
# Closes #21

# This commit closes isses: #21, #20, #18, #17

## Description
I have implemented user accounts based on email and password.
Moreover, registration and login have been created with JWT.
For this, there are three endpoints with the prefix `/api/auth/`.
The endpoints are:

- `login`
- `register`
- `refresh-token`

For this, I needed a Postgre instance so I have added it to docker-compose. As this is a development one I exposed the port to the public so We can connect to DB and check it.

## How to test
Run this with docker-compose - first, check the updated README (You should not need to set the variables with JWT prefix as they should load from `appsetings.development.json`)

First, register user using `/api/auth/register`  (also here should be tested if the user can't be created twice with the same email, password rules are applied, etc.)

Second login using credentials from the previous step `/api/auth/login`

Using the token from this step try to access `/api/greetings/user`  (try if it can be accessed only with this token
Then hit the endpoint  `/api/greetings/helloworld` (this one should be accessible with and without the token as this is a public endpoint)

Afterward, try to request a new token using the refresh token `/api/auth/refresh-token` check if the newly obtained JWT works as well.

For the last step login to the database and set `IsRevoked` for your refresh token to true. Wait 15 minutes (the time till which JWT is valid) and try to request a new token with `/api/auth/refresh-token` . This should now fail.

Lastly use the `/api/auth/logout` to log out. This requires authentication with the current JWT.
This endpoint deletes all currently working refresh tokens for the user. This DOES NOT invalidate current JWT as this is impossible so users can still use it till it is expired.
